magma/array.py

```diff
@@ -606,9 +607,15 @@ def wire(self, o, debug_info):
         o = magma_value(o)
         if not self._check_wireable(o, debug_info):
             return
-        if (self._has_elaborated_children() or
-                o._has_elaborated_children() or
-                self.T.is_mixed()):
+        if (
```

Comment:
let's extract this into a temp named bool, will be more readable. perhaps `should_wire_children`